Solves export of glibc 2.17 secure_getenv because we support down to 2.11

Patches openssl 3.5's usage of secure_getenv from glibc 2.17 to instead
always use the fallback OPENSSL_issetugid(), which essentially does the
same thing on linux, with the only difference that the glibc version makes
decisions on startup, whereas the openssl version does the same check each
time the environment is read.

glibc check: https://sourceware.org/git/?p=glibc.git;a=blob;f=elf/enbl-secure.c;h=9e47526bd3e444e1a19a8ea9fd310b6f47c4db52;hb=HEAD
glibc implementation: https://sourceware.org/git/?p=glibc.git;a=blob;f=stdlib/secure-getenv.c;h=a394eebcf794c1279d66e5bcb71d4b15725e6e5a;hb=HEAD

openssl check: https://github.com/openssl/openssl/blob/openssl-3.5.9/crypto/uid.c

This patch can be removed when glibc 2.17 is the minimum version supported

Author: Patrick Lodder <patricklodder@users.noreply.github.com>

--- a/crypto/getenv.c
+++ b/crypto/getenv.c
@@ -86,16 +86,7 @@
     }
 #endif

-#if defined(__GLIBC__) && defined(__GLIBC_PREREQ)
-#if __GLIBC_PREREQ(2, 17)
-#define SECURE_GETENV
-    return secure_getenv(name);
-#endif
-#endif
-
-#ifndef SECURE_GETENV
     if (OPENSSL_issetugid())
         return NULL;
     return getenv(name);
-#endif
 }
